AnkiDroid/src/main/java/com/ichi2/anki/IntentHandler.kt

```diff
@@ -144,17 +144,27 @@ class IntentHandler : AbstractIntentHandler() {
         }
     }
 
-    private fun handleReviewIntent(intent: Intent) {
+    private fun handleReviewIntent(
+        reloadIntent: Intent,
+        reviewerIntent: Intent,
+    ) {
         val deckId = intent.getLongExtra(ReminderService.EXTRA_DECK_ID, 0)
         Timber.i("Handling intent to review deck '%d'", deckId)
+
         val reviewIntent =
             if (sharedPrefs().getBoolean("newReviewer", false)) {
                 ReviewerFragment.getIntent(this)
             } else {
-                Intent(this, Reviewer::class.java)
+                Intent(this, Reviewer::class.java).apply {
+                    addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                }
             }
         CollectionManager.getColUnsafe().decks.select(deckId)
-        startActivity(reviewIntent)
+        TaskStackBuilder
```

Comment:
@david-allison 

I'd like to get this in. How's this look? Can commit + squash in if it works since PR is single commit

```suggestion
        // Clean the stack out under the reviewer to avoid any incorrect activities / dialogs /
        // data state from prior app usage showing after reviewer exits if going to reviewer directly
        TaskStackBuilder
```

Reply:
Sure!